File: back/src/main/java/csh/back/domain/vote/user/dto/response/VoteUserSaveResponseDto.java
package csh.back.domain.vote.user.dto.response;

import csh.back.domain.vote.user.entity.VoteUser;

public record VoteUserSaveResponseDto(String memberName,
                                      String place,
                                      Integer updateCount) {
    public static VoteUserSaveResponseDto from(VoteUser voteUser) {
        return new VoteUserSaveResponseDto(
                voteUser.getTripMember().getMember().getName(),
                voteUser.getVoteItem().getTripPlace().getName(),
                voteUser.getUpdateCount()
        );
    }
}

File: back/src/main/kotlin/csh/back/domain/vote/user/dto/response/VoteUserSaveResponseDto.kt
package csh.back.domain.vote.user.dto.response

import csh.back.domain.vote.user.entity.VoteUser

data class VoteUserSaveResponseDto(
    val memberName: String,
    val place: String,
    val updateCount: Int,
) {
    companion object {
        @JvmStatic
        fun from(voteUser: VoteUser): VoteUserSaveResponseDto = VoteUserSaveResponseDto(
            memberName = voteUser.tripMember.member.name,
            place = voteUser.voteItem.tripPlace.name,
            updateCount = voteUser.updateCount,
        )
    }
}